manager/controllers/utils.go

```diff
@@ -10,43 +11,92 @@ import (
 	"strings"
 )
 
+const InvalidOffsetMsg = "Invalid offset"
+
 func LogAndRespError(c *gin.Context, err error, respMsg string) {
 	utils.Log("err", err.Error()).Error(respMsg)
-	c.JSON(http.StatusInternalServerError, utils.ErrorResponse{Error: respMsg})
+	c.AbortWithStatusJSON(http.StatusInternalServerError, utils.ErrorResponse{Error: respMsg})
 }
 
 func LogAndRespBadRequest(c *gin.Context, err error, respMsg string) {
 	utils.Log("err", err.Error()).Warn(respMsg)
-	c.JSON(http.StatusBadRequest, utils.ErrorResponse{Error: respMsg})
+	c.AbortWithStatusJSON(http.StatusBadRequest, utils.ErrorResponse{Error: respMsg})
 }
 
 func LogAndRespNotFound(c *gin.Context, err error, respMsg string) {
 	utils.Log("err", err.Error()).Warn(respMsg)
-	c.JSON(http.StatusNotFound, utils.ErrorResponse{Error: respMsg})
+	c.AbortWithStatusJSON(http.StatusNotFound, utils.ErrorResponse{Error: respMsg})
 }
 
-func ApplySort(c *gin.Context, tx *gorm.DB, allowedFields ...string) (*gorm.DB, error) {
+// nolint: prealloc
+func ApplySort(c *gin.Context, tx *gorm.DB, allowedFields ...string) (*gorm.DB, []string, error) {
 	query := c.DefaultQuery("sort", "id")
 	fields := strings.Split(query, ",")
-
+	var appliedFields []string
 	allowedFieldSet := map[string]bool{
 		"id": true,
 	}
 
 	for _, f := range allowedFields {
 		allowedFieldSet[f] = true
 	}
+
 	for _, enteredField := range fields {
 		if strings.HasPrefix(enteredField, "-") && allowedFieldSet[enteredField[1:]] { //nolint:gocritic
 			tx = tx.Order(fmt.Sprintf("%v DESC", enteredField[1:]))
 		} else if allowedFieldSet[enteredField] {
 			tx = tx.Order(fmt.Sprintf("%v ASC", enteredField))
 		} else {
 			// We have not found any matches in allowed fields, return an error
-			return nil, errors.Errorf("Invalid sort field: %v", enteredField)
+			return nil, nil, errors.Errorf("Invalid sort field: %v", enteredField)
 		}
+		appliedFields = append(appliedFields, enteredField)
+	}
+	return tx, appliedFields, nil
+}
+
+// nolint:lll
+func ListCommon(tx *gorm.DB, c *gin.Context, allowedFields []string, path string) (*gorm.DB, *ListMeta, *Links, error) {
```

Comment:
In case of so many return vals maybe it's better to use named return vals. Yes but here it's not so needed because of different types of return values.